src/shapes/group.class.js

```diff
@@ -224,23 +220,53 @@
       object[directive]('deselected', this.__objectSelectionDisposer);
     },
 
+    /**
+     * Checks if object can enter group and logs relevant warnings
+     * @private
+     * @param {fabric.Object} object
+     * @returns
+     */
+    canEnter: function (object) {
+      if (object === this || this.isDescendantOf(object)) {
+        /* _DEV_MODE_START_ */
+        console.warn('fabric.Group: trying to add group to itself, this call has no effect');
```

Comment:
This is wrong.
The call has an effect. At the time this is called `object` is already in the array.
This is bad and causes maximum stack called errors.
That's why I suggest:
1. Throwing
2. move `canEnter` to collection mixin but I fear it will damage perf (more iteration, but I'm not sure it's the case)